Import.

etienneschmitt opened the pull request.

etienneschmitt committed:
Moved star_convex.off to meshes/

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Added some legacy vtk meshes (unstructured grids).

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
added swap_endianness.

Added functions to change the endianness of floating point numbers.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Very beginning of a work to import and export meshes including attributes.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Merge remote-tracking branch 'cgogn/develop' into develop-schmitte

etienneschmitt committed:
using TYPED_TEST in geometry tests.

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
fixed a typo in ihcmap3

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
added swap_endianness functions for signed integers

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Added some documentation in surface_import.

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
do {...} while(false) macro hack

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
some work on reading files (mostly used for loading vtk legacy files for the moment, see polydata.vtk file)

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Merge remote-tracking branch 'cgogn/develop' into develop-schmitte

# Conflicts:
#	cgogn/core/utils/endian.h
#	cgogn/geometry/tests/types/bounding_box_test.cpp

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
added skip_n and reset methods.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

david-cazier commented:
Dans ce code : 
`for (auto & x : *data_)
    x = cgogn::io::internal::swap_endianness(x);`

Est-ce qu'il ne faut pas faire le swap uniquement sur les données lues et pas sur tous le tableau ?

david-cazier commented:
Dans ce code : 
`for (unsigned i = cac.capacity(), end = data_->size(); i < end; ++i)
            cac.template insert_lines<MAP::PRIM_SIZE>();`

Est-ce que i ne doit pas être incrémenté de PRIM_SIZE ?

david-cazier commented:
Dans la méthode : 
`std::unique_ptr<DataIOGen<MAP>> DataIOGen<MAP>::newDataIO(const std::string type_name)`

On peut se passer des accolades avec : 
`if () 
 else if ()
 else if ()`

david-cazier commented:
Ca me semble très compliqué, ou alors j'ai loupé un truc.

Pourquoi la méthode read_n ne travaille pas directement sur le ChunkArray avec un tableau interne (qui joue le rôle de data_) en ajoutant à la fin ce qui est fait dans to_chunck_array().

En pratique un DataIO c'est un chunck_array d'un seul bloc non ? Du coup je ne vois pas l'intérêt de faire une nouvelle classe, mais s'il existe je suis prêt à être convaincu :)

etienneschmitt commented:
Ok pour les 2 premiers messages, les coquilles sont corrigées.
Concernant DataIO, j'aimerais bien savoir comment tu t'y prendrais pour importer des données dont le type t'est donné dans une chaîne de caractère. Si tu proposes plus simple que mon code on adoptera ta solution :-)

etienneschmitt committed:
added "base64" thirdparty

It is necessary to decode/encode in base64. Used within vtk XML files.

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
fixed minor bugs in DataIO::read_n and splitted surface_import in several subclasses.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
restored this->clear() in SurfaceImport

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
added 2 meshes (vtk legacy unstructured grid of voxels)

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Added MeshImportGen class.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
work on volume import.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
DataIOGen, DataIO -> DataInputGen, DataInput

Added the possibilty of forcing a conversion of the Data after having read it from the file.
This is necessary when we want to have a position attribute of type Vec3d and the data in the files are of type Vec3f.

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
make_unique.h -> unique_ptr.h

Added dynamic_cast_unique_ptr function.

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Removed base64 thirdparty

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
added base64_decode function.

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
fixed a bug when parsing std::(u)int8_t from a stringstrim and added DataInputGen::simplify method

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
vtu import (both ascii and binary) is now working pretty well.

Signed-off-by: Etienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Fixed Format -> format in liverHexa.vtu.
Added binary version of the "nine hexas" mesh.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
fixed some details when importing VTU file.
Every vertex or cell attribute is now added to the map during the mesh loading.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
testing zlib decompression of vtk xml meshes.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

pierrekraemer commented:
@etienneschmitt tu préviens quand tu as terminé pour merger cette PR ?

etienneschmitt committed:
zlib_decompress function is only available if Zlib is found.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
added some doc about dynamic_cast_unique_ptr

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
small changes in zlib_decompress and base64_decode.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
added a small example of a compressed binary vtu mesh.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Fixed a bug when swapping the face or volume chunk array container during the loading.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Added  class CharArrayBuffer and ImemoryStream to avoid unnecessary copies.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

pierrekraemer commented:
En conflit depuis le merge de #132
Tu corriges et on merge ou tu veux encore ajouter des nouvelles choses ?

etienneschmitt committed:
added operator<< for Vec_T

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
fixed a bug when detecting endianness in DataInput.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
base64_decode and zlib_decompress now take a const char* instead of a std::string.

Also added move constructor / assignment for CharArrayBuffer and IMemoryStream

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
added mesh heart_zlib.vtu

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Merge remote-tracking branch 'cgogn/develop' into develop-schmitte

# Conflicts:
#	cgogn/geometry/types/geometry_traits.h

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>
